euphonic/spectra.py

```diff
@@ -821,16 +819,44 @@ def __getitem__(
             self, item: Union[Integral, slice, Sequence[Integral], np.ndarray]
     ):  # noqa: F811
         self._validate_item(item)
-        init_kwargs = {
-            self._spectrum_data_name(): self._get_spectrum_data()[item, :],
-            "x_tick_labels": self.x_tick_labels,
-            "metadata": self._get_item_metadata(item)
-                       } | self._get_bin_kwargs()
 
         if isinstance(item, Integral):
-            return self._item_type(**init_kwargs)
+            spectrum = self._item_type.__new__(self._item_type)
+        else:
+            spectrum = self.__new__(type(self))
+
+        self._set_item_data(spectrum, item)
 
-        return type(self)(**init_kwargs)
+        spectrum.x_tick_labels = self.x_tick_labels
+        spectrum.metadata = self._get_item_metadata(item)
+
+        return spectrum
+
+    def _set_item_data(
+            self,
+            spectrum: Spectrum,
+            item: Union[Integral, slice, Sequence[Integral], np.ndarray]
+    ) -> None:
+        """Write axis and spectrum data from self to Spectrum
+
+        This is intended to set attributes on a 'bare' Spectrum created with
+        __new__() from the parent SpectrumCollection.
+        """
+
+        for axis in self._bin_axes:
+            setattr(spectrum, f"_{axis}_data",
+                    getattr(self, f"_{axis}_data").copy())
+            setattr(spectrum, f"_internal_{axis}_data_unit",
+                    getattr(self, f"_internal_{axis}_data_unit"))
+            setattr(spectrum, f"{axis}_data_unit",
+                    getattr(self, f"{axis}_data_unit"))
```

Comment:
```suggestion
            for prop in ("_{}_data", "_internal_{}_data_unit", "{}_data_unit"):
                curr_prop = prop.format(axis)
                setattr(spectrum, curr_prop, getattr(self, curr_prop).copy())
```
?

Reply:
It was mostly just an "is this more legible" thing anyway. It's fine as is, just a little visually noisy.

Reply:
Maybe! Looks like `copy` on str is a no-op as it should be:

```python
In [6]: def string_assign():
   ...:     for _ in range(10000):
   ...:         b = x
   ...:

In [7]: def string_copy():
   ...:     for _ in range(10000):
   ...:         b = copy(x)
   ...:

In [8]: %timeit string_assign
4.04 ns ± 0.0335 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)

In [9]: %timeit string_copy
4.03 ns ± 0.0177 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```

Reply:
The overhead on an array seems tolerable too. (It goes through a couple of lines of logic to find the `__copy__` method then delegates... https://github.com/python/cpython/blob/3.13/Lib/copy.py)

```python
In [12]: dat = np.random.random((1000, 1000))

In [13]: def numpy_copy():
    ...:     for _ in range(10000):
    ...:         b = dat.copy()
    ...:

In [14]: def copy_numpy():
    ...:     for _ in range(10000):
    ...:         b = copy(dat)
    ...:

In [15]: %timeit numpy_copy()
3.1 s ± 64.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [16]: %timeit copy_numpy()
3.15 s ± 131 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [17]:
```